### DISCUSSION TITLE
esp32-s2 serial to TCP bridge commands unknown

### DISCUSSION THREAD

**Discussion by frgazor** [arendst/Tasmota#21567] [Q&A] [answered]
Hi,

I'm trying to build custom binary with serial to TCP bridge to communicate with my 3d printer over wifi.  In my user_config_override I have #define USE_TCP_BRIDGE as described in documentation. Everything compiles fine and adds ~3kb to binary size when USE_TCP_BRIDGE is enabled. All my other custom compiled options, like tm1637 display support, work fine with this build. However, serial to TCP bridge shows no signs of life and TCPBaudRate and TCPStart commandfs return "command unknown". GPIO pins are assigned to RX and TX.

I have tried building my fw using tasmota 13.4 and latest development version from github but no luck.

Any ideas? Thanks

**Comment by sfromis** [comment, marked as answer]
You said that you had TX/RX pins assigned, instead of telling what you configured them as, like if you followed the docs:
https://tasmota.github.io/docs/Serial-to-TCP-Bridge/#configuration

**Comment by frgazor** [comment]
Ah... I obviously just needed someone to make me read the fine docs again. I have read documentation numerous times and failed to see the difference between TCP Rx/Tx and Serial Rx/Tx that I have selected in configuration. It's working now.
